client/js/components/StatsLeaderboard.vue

```diff
@@ -0,0 +1,359 @@
+        return [];
+      }
+      const res = [
+        { text: 'User', value: 'username' },
+        { text: 'Rank', value: 'rank' },
+        { text: 'Season Total', value: 'week_total' },
+      ];
+      // Add headers for each week
+      for (const weekNum of this.selectedWeeks) {
+        res.push({
+          text: `Week ${weekNum}`,
+          value: `week_${weekNum}`,
+        });
+      }
+      return res;
+    },
+    tableRows() {
+      if (!this.season || !this.season.rankings || this.season.rankings.length === 0) {
+        return [];
+      }
+      return this.season.rankings.map((playerStats, index) => {
+        const playerWins = this.playerWins[index];
+        const playerScores = this.playerScores[index];
+        const res = {
+          username: playerStats.username,
+          week_total_wins: playerWins.total,
+          week_total_points: playerScores.total,
+          week_total: playerScores.total,
+          rank: this.rank(playerScores.total),
+        };
+        for (const weekNum in playerStats.matches) {
+          res[`week_${weekNum}`] = playerScores[weekNum];
+          res[`week_${weekNum}_wins`] = playerWins[weekNum];
+          res[`week_${weekNum}_points`] = playerScores[weekNum];
+        }
+        return res;
+      });
+    },
+    playerWins() {
+      if (!this.season || !this.season.rankings || this.season.rankings.length === 0) {
+        return [];
+      }
+      return this.season.rankings.map(playerStats => {
+        const res = { total: 0 };
+        const playerMatches = playerStats.matches;
+        for (const weekNum of this.weeks.map(week => week.value)) {
+          const matchesThisWeek = playerMatches[weekNum] || [];
+          const wins = matchesThisWeek.filter(match => match.result === Result.WON);
+          res.total += wins.length;
+          res[`${weekNum}`] = wins.length;
+        }
+        return res;
+      });
+    },
+    playerScores() {
+      if (!this.season || !this.season.rankings || this.season.rankings.length === 0) {
+        return [];
+      }
+      return this.playerWins.map(playerWins => {
+        const res = { total: 0 };
+        for (const weekNum in playerWins) {
+          if (weekNum != 'total') {
+            let pointsThisWeek = 0;
+            if (this.topWinCountsPerWeek[weekNum].first === 0) {
+              pointsThisWeek = 0;
+            } else if (playerWins[weekNum] === this.topWinCountsPerWeek[weekNum].first) {
+              pointsThisWeek = 5;
+            } else if (playerWins[weekNum] === this.topWinCountsPerWeek[weekNum].second) {
+              pointsThisWeek = 4;
+            } else if (playerWins[weekNum] === this.topWinCountsPerWeek[weekNum].third) {
+              pointsThisWeek = 3;
+            } else if (playerWins[weekNum] > 0) {
+              pointsThisWeek = 1;
+            }
+            res[weekNum] = pointsThisWeek;
+            res.total += pointsThisWeek;
+          }
+        }
+        return res;
+      });
```

Comment:
I really prefer to avoid nesting ifs when it can be helped, it ensures the code stays easily readable. Are you able to filter the weekNums before they go in to the loop? Something like `for (const weekNum in playerWins.filter(pw.weekNum != 'total')) {`, then you wouldn't need the nested if.

Reply:
Yes that's cleaner (though seemingly less performant since we loop through the playerWins an extra time. But I'm comfortable prioritizing legibility

Reply:
Oh wait I think we hit a gotcha here because we're iterating over object properties with for...in not over an array with for...of

Reply:
got it like
```
    const { total, ...weeks } = playerWins;
    for (const weekNum in { ...weeks }) {
```